### PR TITLE
Reload user settings when changed externally

### PR DESCRIPTION
**Pull request by lf94**

### CODE REVIEW

File: src/hooks/useFileSystemWatcher.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ type Path = string
 // watcher.addListener(() => { ... }).
 
 export const useFileSystemWatcher = (
-  callback: (path: Path) => void,
+  callback: (path: Path) => any | Promise<any>,
```
Comment (jtran) [inline src/hooks/useFileSystemWatcher.tsx:14]: I don't think there's any need for this change. It disables the type checker for no reason.
```suggestion
  callback: (path: Path) => void,
```

Comment (lf94) [reply src/hooks/useFileSystemWatcher.tsx:14]: There is, go and try it with void :)

Comment (lf94) [reply src/hooks/useFileSystemWatcher.tsx:14]: Admittedly it was pointless of me to do any | Promise<any>. I should have just been any...

Comment (lf94) [reply src/hooks/useFileSystemWatcher.tsx:14]: I think it's silly to assume `void` is correct on behalf of the user. We don't care what the callback returns. Hm. I wonder if maybe <T>(path) => T would have worked

Comment (lf94) [reply src/hooks/useFileSystemWatcher.tsx:14]: Ok after like another 5 seconds I'm telling myself there is no way to get the return value anyway. I should have just done void | Promise<void> :facepalm:

Comment (jtran) [reply src/hooks/useFileSystemWatcher.tsx:14]: As long as there's no `any`, I'm happy 🙂

**Comment by nadr0** [comment]
I ran this locally and ran `vim ~/.config/zoo-modeling-app/settings.toml ` to edit the settings file and the refresh only worked once then it stopped working when I edited the file for the N+1 times.

**Comment by lf94** [comment]
Aw shit eh, I swear it worked more than once here. Lemme see...

**Comment by lf94** [comment]
@nadr0 I think it depends on the setting. Some settings are more eager to change than others:


https://github.com/user-attachments/assets/8730d938-12e6-44bc-9b70-79e338119238

File: src/hooks/useFileSystemWatcher.tsx (hunk 1 above)
Comment (lf94) [reply src/hooks/useFileSystemWatcher.tsx:14]: Actually it doesn't work with `void | Promise<void>` either. The linter is dumb.

out

**Comment by nadr0** [comment]
https://github.com/user-attachments/assets/5d9943f5-59c6-412a-ae19-358b4665d7af

I am unable to swap infinitely back and forth between light and dark?

File: src/hooks/useFileSystemWatcher.tsx (hunk 1 above)
Comment (jtran) [reply src/hooks/useFileSystemWatcher.tsx:14]: #4098

**Comment by nadr0** [comment]
I tested this new fix with `chokidar` and it works on my localhost on ubuntu. The native `node:fs` implementation of watch had some issues. That library has resolved those issues.

File: src/preload.ts [hunk 2]
```diff
@@ -23,36 +24,19 @@ const isMac = os.platform() === 'darwin'
...
 
-const watchFileOn = (
-  path: string,
-  callback: (eventType: string, path: string) => void
-) => {
-  const watcher = fsSync.watch(path)
-  watcher.on('change', callback)
-  fsWatchListeners.set(path, { watcher, callback })
+const watchFileOn = (path: string, callback: (path: string) => void) => {
+  const watcher = chokidar.watch(path)
```
Comment (nadr0) [inline src/preload.ts:30]: Should we do a check to see if the `fsWatchListeners` already has this path in some edge case so it does not do `chokidar.watch` twice on something?

Comment (lf94) [reply src/preload.ts:30]: Wouldn't hurt! Even though I can't think of a case when, never a bad idea to be safer.

Comment (lf94) [reply src/preload.ts:32]: Done

**Review by nadr0 — APPROVED** [review]
Thanks for addressing those PR comments and fixing the bugs.

I tested locally on linux and was able to spam change light/dark from `vim` when directly editing the settings file.

Great work on this!

**Pull request merged**